ksql-engine/src/main/java/io/confluent/ksql/util/SerDeUtil.java

```diff
@@ -67,24 +69,28 @@
     return Serdes.serdeFrom(genericRowSerializer, genericRowDeserializer);
   }
 
-  public static Serde<GenericRow> getGenericRowAvroSerde(final Schema schema) {
+  public static Serde<GenericRow> getGenericRowAvroSerde(final Schema schema, final KsqlConfig ksqlConfig) {
     Map<String, Object> serdeProps = new HashMap<>();
     String avroSchemaString = new MetastoreUtil().buildAvroSchema(schema, DdlConfig.AVRO_SCHEMA);
     serdeProps.put(KsqlGenericRowAvroSerializer.AVRO_SERDE_SCHEMA_CONFIG, avroSchemaString);
 
-    final Serializer<GenericRow> genericRowSerializer = new KsqlGenericRowAvroSerializer(schema);
+    SchemaRegistryClient schemaRegistryClient = SchemaRegistryClientFactory.getSchemaRegistryClient(ksqlConfig.getString(KsqlConfig
+                                                                                                                             .SCHEMA_REGISTRY_URL_PROPERTY));
+    final Serializer<GenericRow> genericRowSerializer = new KsqlGenericRowAvroSerializer(schema,
+                                                                                         schemaRegistryClient, ksqlConfig);
     genericRowSerializer.configure(serdeProps, false);
 
     final Deserializer<GenericRow> genericRowDeserializer =
-        new KsqlGenericRowAvroDeserializer(schema);
+        new KsqlGenericRowAvroDeserializer(schema, schemaRegistryClient);
     genericRowDeserializer.configure(serdeProps, false);
 
     return Serdes.serdeFrom(genericRowSerializer, genericRowDeserializer);
   }
 
-  public static Serde<GenericRow> getRowSerDe(final KsqlTopicSerDe topicSerDe, Schema schema) {
+  public static Serde<GenericRow> getRowSerDe(final KsqlTopicSerDe topicSerDe, Schema schema,
+                                              KsqlConfig ksqlConfig) {
     if (topicSerDe instanceof KsqlAvroTopicSerDe) {
-      return SerDeUtil.getGenericRowAvroSerde(schema);
+      return SerDeUtil.getGenericRowAvroSerde(schema, ksqlConfig);
```

Comment:
see not above about having a method on `KsqlTopicSerde` rather than `if(...) else if(...) else`